Update manual download section to add GitHub Actions and update Cirrus CI section. (#239)
pulsar-edit/pulsar-edit.github.io: docs/download.md

stick to the **master** branch releases for the most stable ones unless you know
exactly what you are looking for.

We currently build on both GitHub Actions and Cirrus CI:

- GitHub Actions is used for building the majority of our binaries and builds
  are run on every commit or PR.
  - Direct artifact download is available only for those with at least read
    permission on the repository. All artifacts are therefore automatically
    uploaded to the [pulsar-edit/pulsar-rolling-releases](https://github.com/pulsar-edit/pulsar-rolling-releases/releases)
    repository for access by anyone.
- Cirrus CI is _only_ used for building Apple silicon (macOS M1/M2) and ARM

>   Linux binaries. These are run less frequently (every other day).

Reviewer: ```suggestion
  Linux binaries. These are run less frequently (every Monday, Wednesday, and Friday).
```
